llrt_modules/src/modules/process.rs

```diff
@@ -14,3 +23,152 @@ pub fn get_platform() -> &'static str {
         _ => platform,
     }
 }
+
+fn cwd() -> String {
+    env::current_dir().unwrap().to_string_lossy().to_string()
+}
+
+pub fn get_arch() -> &'static str {
+    let arch = env::consts::ARCH;
+
+    match arch {
+        "x86_64" | "x86" => return "x64",
+        "aarch64" => return "arm64",
+        _ => (),
+    }
+
+    arch
+}
+
+fn hr_time_big_int(ctx: Ctx<'_>) -> Result<BigInt> {
+    let now = Utc::now().timestamp_nanos_opt().unwrap_or_default() as u64;
+    let started = TIME_ORIGIN.load(Ordering::Relaxed) as u64;
+
+    let elapsed = now - started;
+
+    BigInt::from_u64(ctx, elapsed)
+}
+
+fn hr_time(ctx: Ctx<'_>) -> Result<Array<'_>> {
+    let now = Utc::now().timestamp_nanos_opt().unwrap_or_default() as u64;
+    let started = TIME_ORIGIN.load(Ordering::Relaxed) as u64;
+    let elapsed = now - started;
+
+    let seconds = elapsed / 1_000_000_000;
+    let remaining_nanos = elapsed % 1_000_000_000;
+
+    let array = Array::new(ctx)?;
+
+    array.set(0, seconds)?;
+    array.set(1, remaining_nanos)?;
+
+    Ok(array)
+}
+
+fn exit(code: i32) {
+    std::process::exit(code)
+}
+
+fn env_proxy_setter<'js>(
+    target: Object<'js>,
+    prop: Value<'js>,
+    value: Coerced<String>,
+) -> Result<bool> {
+    target.set(prop, value.to_string())?;
+    Ok(true)
+}
+
+pub fn init(ctx: &Ctx<'_>) -> Result<()> {
```

Comment:
I think down the line we need to allow module settings to say not allow the module to load the process env.